lib/src/widgets/custom_credit_card.dart

```diff
@@ -0,0 +1,254 @@
+                      onSaved: (value) {
+                        _cardData.cvc = value ?? '';
+                        widget.onCreditCardFormChange(
+                            _cardData, _isValidForm());
+                      }),
+                ),
+              ],
+            )
+          else ...[
+            CardFormField(
+              inputDecoration: buildInputDecoration(
+                hintText: '${widget.locale.expiry} (MM / YY)',
+              ),
+              inputFormatters: [
+                FilteringTextInputFormatter.digitsOnly,
+                LengthLimitingTextInputFormatter(4),
+                CardMonthInputFormatter(),
+              ],
+              validator: (String? input) =>
+                  CardUtils.validateDate(input, widget.locale),
+              onSaved: (value) {
+                List<String> expireDate = CardUtils.getExpiryDate(value!);
+                if (expireDate.length == 2) {
+                  _cardData.month = expireDate.first;
+                  _cardData.year = expireDate[1];
+                }
+                widget.onCreditCardFormChange(_cardData, _isValidForm());
+              },
+            ),
+            CardFormField(
+                inputDecoration: buildInputDecoration(
+                  hintText: widget.locale.cvc,
+                ),
+                inputFormatters: [
+                  FilteringTextInputFormatter.digitsOnly,
+                  LengthLimitingTextInputFormatter(4),
+                ],
+                validator: (String? input) =>
+                    CardUtils.validateCVC(input, widget.locale),
+                onSaved: (value) {
+                  _cardData.cvc = value ?? '';
+                  widget.onCreditCardFormChange(_cardData, _isValidForm());
+                }),
+          ],
+        ],
+      ),
+    );
+  }
+}
+
+class CardFormField extends StatelessWidget {
+  final void Function(String?)? onSaved;
+  final String? Function(String?)? validator;
+  final TextInputType keyboardType;
+  final TextInputAction textInputAction;
+  final List<TextInputFormatter>? inputFormatters;
+  final InputDecoration? inputDecoration;
+
+  const CardFormField({
+    Key? key,
+    required this.onSaved,
+    this.validator,
+    this.inputDecoration,
+    this.keyboardType = TextInputType.number,
+    this.textInputAction = TextInputAction.next,
+    this.inputFormatters,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 8.0),
+      child: TextFormField(
+          keyboardType: keyboardType,
+          textInputAction: textInputAction,
+          decoration: inputDecoration,
+          validator: validator,
+          onChanged: (value) {
+            onSaved?.call(value);
+          },
+          inputFormatters: inputFormatters),
+    );
+  }
+}
+
+String showAmount(int amount, Localization locale) {
```

Comment:
It should be inside the class and private
if it uses in other files you can create a different class(`CardUtils` or etc) otherwise move it to the `CustomCreditCard`

Same for the below code